### ISSUE TITLE
[BUG] payment gets scrubbed 2x times

### ISSUE THREAD

**Issue by inf0matics** [lnbits/scrub#7]
### Description

After updating scrubs plugin from 0.3 to 0.3.1 funds get scrubbed twice. I switched back to 0.3 and it worked as expected.

<img width="662" alt="Screenshot 2024-02-13 at 11 16 17" src="https://github.com/lnbits/scrub/assets/3327667/73300be7-13d0-4f49-87c6-99c0e9f268d4">

### Parameters
- LNbits version: 0.11.1
- Sending Wallet: different lnbits wallet on same lnbits, getalby

**Comment by dni** [comment]
thanks for the report, we have a issue, when upgrading ext at runtime, can you try again to update and restart lnbits via the CLI and not the admin ui restart? see if it still scrubing twice

**Comment by inf0matics** [comment]
I did a restart of lnbits itself, after upgrading to scrubs to 0.3.1. --> same result: scrubbed 2x times

We have the same setup as "test/stage setup" ... with 0.3.1. it gets scrubbed only 1x times 🫣

**Comment by inf0matics** [comment]
Ok we have the same problem with 0.3.

**Comment by inf0matics** [comment]
mhm ok it gets even wired ... split payments does it too 🫣 and we have the some problem on our staging system too (which is an exact copy)

![Screenshot 2024-02-14 at 13 55 36](https://github.com/lnbits/scrub/assets/3327667/57b45cfd-8c54-438b-8dfa-745133914f5a)

**Comment by inf0matics** [comment]
Logs:

```text
2024-02-14 13:17:27.21 | INFO | lnbits.tasks:invoice_listener:137 | > got a payment notification
2024-02-14 13:17:27.22 | DEBUG | lnbits.core.tasks:dispatch_api_invoice_listeners:139 | sending invoice paid event to db3XXXXXXXXXXXXX9fc_d2c9592f
2024-02-14 13:17:27.22 | DEBUG | lnbits.core.views.api:subscribe_wallet_invoices:477 | sse listener: payment received
2024-02-14 13:17:27.30 | INFO | 80.109.220.78:0 - "GET /api/v1/wallet HTTP/1.1" 200
2024-02-14 13:17:27.37 | INFO | 80.109.220.78:0 - "GET /api/v1/payments/paginated?limit=10&offset=0&sortby=time&direction=desc HTTP/1.1" 200
2024-02-14 13:17:27.43 | INFO | 80.109.220.78:0 - "GET /api/v1/wallet HTTP/1.1" 200
2024-02-14 13:17:28.22 | WARNING | lnbits.utils.exchange_rates:fetch_price:255 | Failed to fetch Bitcoin price for EUR from Bitstamp: 
2024-02-14 13:17:28.22 | WARNING | lnbits.utils.exchange_rates:fetch_price:255 | Failed to fetch Bitcoin price for EUR from Coinbase: 
2024-02-14 13:17:28.22 | WARNING | lnbits.utils.exchange_rates:fetch_price:255 | Failed to fetch Bitcoin price for EUR from CoinMate: 
2024-02-14 13:17:28.30 | WARNING | lnbits.utils.exchange_rates:fetch_price:255 | Failed to fetch Bitcoin price for EUR from Exir: Client error '400 Bad Request' for url 'https://api.exir.io/v1/ticker?symbol=btc-eur'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
2024-02-14 13:17:28.30 | DEBUG | lnbits.core.services:calculate_fiat_amounts:97 | Calculated fiat amounts wallet.id='db3XXXXXXXXXXXXX9fc' amount=5.0 currency=None: extra={'tag': 'scrubed', 'wallet_fiat_currency': 'EUR', 'wallet_fiat_amount': 0.002, 'wallet_fiat_rate': 2077.099924551968}
2024-02-14 13:17:28.30 | DEBUG | lnbits.core.services:pay_invoice:253 | creating temporary payment with id 6dbefcac8XXXXXXXXXXXXX3675bab981ae9e5afd1
2024-02-14 13:17:28.31 | DEBUG | lnbits.core.services:pay_invoice:297 | backend: sending payment 6dbefcac8XXXXXXXXXXXXX3675bab981ae9e5afd1
2024-02-14 13:17:28.37 | DEBUG | lnbits.core.services:calculate_fiat_amounts:97 | Calculated fiat amounts wallet.id='db3XXXXXXXXXXXXX9fc' amount=5.0 currency=None: extra={'tag': 'scrubed', 'wallet_fiat_currency': 'EUR', 'wallet_fiat_amount': 0.002, 'wallet_fiat_rate': 2077.099924551968}
2024-02-14 13:17:28.37 | DEBUG | lnbits.core.services:pay_invoice:253 | creating temporary payment with id afd12e18c9fcd58XXXXXXXXXXXXXbd5e7881c05ce2392c9a0
2024-02-14 13:17:28.38 | DEBUG | lnbits.core.services:pay_invoice:297 | backend: sending payment afd12e18c9fcd58XXXXXXXXXXXXXbd5e7881c05ce2392c9a0
2024-02-14 13:17:36.21 | DEBUG | lnbits.core.services:pay_invoice:310 | backend: pay_invoice finished afd12e18c9fcd58XXXXXXXXXXXXXbd5e7881c05ce2392c9a0
2024-02-14 13:17:36.21 | DEBUG | lnbits.core.services:pay_invoice:313 | updating payment afd12e18c9fcd58XXXXXXXXXXXXXbd5e7881c05ce2392c9a0
2024-02-14 13:17:36.22 | DEBUG | lnbits.core.services:pay_invoice:329 | payment successful afd12e18c9fcd58XXXXXXXXXXXXXbd5e7881c05ce2392c9a0
2024-02-14 13:17:36.29 | DEBUG | lnbits.core.services:pay_invoice:310 | backend: pay_invoice finished 6dbefcac8XXXXXXXXXXXXX3675bab981ae9e5afd1
2024-02-14 13:17:36.29 | DEBUG | lnbits.core.services:pay_invoice:313 | updating payment 6dbefcac8XXXXXXXXXXXXX3675bab981ae9e5afd1
2024-02-14 13:17:36.29 | DEBUG | lnbits.core.services:pay_invoice:329 | payment successful 6dbefcac8XXXXXXXXXXXXX3675bab981ae9e5afd1
```

Any ideas?

**Comment by inf0matics** [comment]
I think it's maybe a bug in lnbits? The Splitpayments extension is not enabled for that user (but it is installed on this instance) ... and it move funds?

![Screenshot 2024-02-14 at 17 03 50](https://github.com/lnbits/scrub/assets/3327667/bcc4496f-5d57-44ee-94d1-5fadf4ecba37)

**Comment by arbadacarbaYK** [comment]
Relates to https://github.com/lnbits/lnbits/pull/2250

**Comment by inf0matics** [comment]
After discussion with @dni I tried to make sure the docker container could resolve localhost to 127.0.0.1

I was using the docker images:
- lnbits/lnbits:0.12.1
- lnbitsdocker/lnbits-legend:0.11.3

The following was working for me & I tested it with both images

```bash
docker exec -it <lnbits-image-name> /bin/sh
apt install libnss-myhostname
exit

docker restart <lnbits-image-name>
# OR restart lnbits via admin ui
```

> [!CAUTION]
> This is a temporary fix, because if you remove the container, you need to apply the fix again

**Comment by motorina0** [comment]
Should be fixed with release 0.12.2

**Comment by inf0matics** [comment]
Just tested it on our dev and production environment. Thx it's working!

I would have closed the issue now 😉
